### ISSUE TITLE
confusing:  Users of the application have to separately install the .NET runtime.

### ISSUE THREAD

**Issue by Rick-Anderson** [dotnet/docs#41489]
### Type of issue

Other (describe below)

### Description

Publishing your app as framework-dependent produces an application that includes only your application itself and its dependencies. Users of the application have to separately install the .NET runtime.

Most of our customers are publishing to Azure (or another cloud) that contains the .NET runtime. We need better wording and should mention when you pub Azur App Svc framework-dependent, Azure has the runtime installed.

@adegeo @IEvangelist  please suggest better wording

After we agree on the wording I'll do a "you" ectomy. It reads much better to say

Publishing _the_ app as self-contained 
than
Publishing _your_ app as self-contained 
 
The doc is shorter an more accurate. The zillions of consultants that read this don't need to roll their eyes by all the "your"'s.

### Page URL

https://learn.microsoft.com/en-us/dotnet/core/deploying/#framework-dependent-deployments-fdd

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/core/deploying/index.md

### Document Version Independent Id

89967250-8c3b-1999-078f-94ad7518ed49

### Article author

@adegeo

### Metadata

* ID: 85eba478-0374-bb9b-8fa7-d5cf7d520603 
* Service: **dotnet-fundamentals**

**Comment by adegeo** [comment]
So are you thinking something like 

> Publishing your app as framework-dependent produces an application that includes only your application itself and its dependencies. Users of the application have to separately install the .NET runtime. If you plan to run the app in the cloud, make sure that the target framework is already available on the host.

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson So are you thinking something like
> 
> > Publishing your app as framework-dependent produces an application that includes only your application itself and its dependencies. Users of the application have to separately install the .NET runtime. If you plan to run the app in the cloud, make sure that the target framework is already available on the host.

That's too nebulous. 
_Publishing your app as framework-dependent produces an application that includes only your application itself and its dependencies._  **Users of the application have to separately install the .NET runtime.**  The bold might be technically right but it reads like on Azure with a released version of .NET, you have to **separately install the .NET runtime.** 

We should feature Azure first as that's where are customers are going and where we want to encourage them to go. I wouldn't start out telling them you have to **separately install the .NET runtime.**  I'd start out saying something like

Publishing your app as framework-dependent produces an application that includes only your application and its dependencies, excluding the .NET runtime. If you plan to run the app on Azure App Services and are using a released version of .NET, the .NET runtime is already installed on the service, so you don't need to install it separately.

AWS has eons of latency and they don't bundle the runtime for doggie years after the release. I'm not too worried about explaining AWS and other clouds.

**Comment by adegeo** [comment]
@Rick-Anderson I don't agree with that direction (but I'll welcome feedback from the product group). We have tons of customers doing desktop and console apps. This documentation is speaking to .NET in a generic sense from the product's point-of-view. If we want to improve the content with some sections talking about cloud stuff, I think that's fine, but I don't think re-tuning the entire doc to be cloud-first is the appropriate direction.

I would expect the user to actually research their cloud service and what it has prior to deploying something to.

Adding @richlander @KathleenDollard who helped drive this article.

**Comment by Rick-Anderson** [comment]
> @Rick-Anderson but I don't think re-tuning the entire doc to be cloud-first is the appropriate direction.

I don't want to re-tune the entire doc, just this section, just the following minor change

***Publishing your app as framework-dependent produces an application that includes only your application and its dependencies, excluding the .NET runtime. If you plan to run the app on Azure App Services and are using a released version of .NET, the .NET runtime is already installed on the service, so you don't need to install it separately.***

> I would expect the user to actually research their cloud service and what it has prior to deploying something to.

Why should they have to research and hunt around to find out you don't typically need to install .NET?

When you publish an ASP.NET Core app to Azure, a message is displayed "publishing framework-dependent" When the user googles "framework-dependent", this is what they get, the problematic text:

<img width="464" alt="image" src="https://github.com/dotnet/docs/assets/3605364/3488bd54-099c-4aae-81be-df8fdfa4eda5">

[**Publish framework-dependent**](https://learn.microsoft.com/en-us/dotnet/core/deploying/#publish-framework-dependent)

Apps published as framework-dependent are cross-platform and don't include the .NET runtime. ***The user of your app is required to install the .NET runtime.***

>***The user of your app is required to install the .NET runtime.***

That's so confusing to new users. Jane publishes a web app but the users of her app have to install the .NET runtime?

**Comment by richlander** [comment]
I would avoid mentioning cloud or Azure at all in this little section. Application publishing is very generic.

How about this:

> Publishing an application project as a framework-dependent application produces a set of binaries that includes app code and dependencies but not the .NET runtime. The app can be run in any environment or machine where the correct version of the .NET runtime has been installed.

**Comment by adegeo** [comment]
I think that users would be better served with technology-specific publishing articles. This would help SEO much more than putting blurbs in the content that doesn't get picked up:

- How to publish an ASP.NET Core app
- How to deploy an ASP.NET Core service on Azure
- Publish a Windows Forms app
- .NET Aspire deployments
- Tutorial: Publish a .NET console application using Visual Studio Code

**Comment by richlander** [comment]
Now that I look at the article, not just the text here, I think the page needs to be re-written. It was written for a different time, back with .NET Core 1.0 and before we had a good cloud story.

You can see that in this image. The TOC says "deployment models" and the articles says "publishing overview". Those are not the same thing and are the crux of this discussion.

<img width="869" alt="image" src="https://github.com/dotnet/docs/assets/2608468/e611dbbe-db07-4e91-bfe4-0941c42a293f">

Ideally, the article would lead with a discussion of common deployment targets and how it is straightforward to produce apps and other artifacts (like containers) for them. And then the rest would be links to other content, along the lines of what @adegeo suggested. This approach would also enable us to talk about cloud services sooner, per what @Rick-Anderson suggested. We'd then move all of the framework-dependent and friends content to a different topic.

I've always found this topic to be an abrupt intro into the general topic of deployment. I've never found a good way of describing that.

I'm happy to help on this if you want to share a branch.

**Comment by adegeo** [comment]
Great points everyone. Each article in this area has kind of evolved independently and there hasn't been an overall story. Perhaps we should start with planning a high level overview of what we want the area to describe.

**Comment by adegeo** [comment]
Tagging @IEvangelist who has done some work on this article recently.

**Comment by IEvangelist** [comment]
> Tagging @IEvangelist who has done some work on this article recently.

Just read through all of the issue comments, I agree that this seems a bit dated, and each area has evolved independently. It makes sense to refocus on this area all up and plan on updates.

**Comment by richlander** [comment]
I think to start, we can just move the current content to its own topic and write a new intro. That should help us determine what's actually needed. I can start that if folks want.

**Comment by adegeo** [comment]
I'll take a look at redesigning and rewriting this content in Dec. Thanks everyone!

**Comment by adegeo** [comment]
Pushed to Jan due to priority.

**Comment by adegeo** [comment]
Due to quite a few high priority items, I'm moving this out to April.

**Comment by adegeo** [comment]
@richlander I created PR #47079 which takes a fresh look at the topic. It merges both the Visual Studio and CLI versions of the doc into a single overview article. Most of the deploying terms were removed except for the titles of things like `Framework-dependent deployment`. It's still in a "deploy" folder but I think that will be a different issue to solve later (the redesigning the entire area for publishing and deploying).
